pkg/controller/container-runtime-config/container_runtime_config_controller.go

```diff
@@ -548,16 +555,34 @@ func (ctrl *Controller) syncImageConfig(key string) error {
 	// Fetch the ImageConfig
 	imgcfg, err := ctrl.imgLister.Get("cluster")
 	if errors.IsNotFound(err) {
-		glog.V(2).Infof("ImageConfig doesn't exist or has been deleted")
+		glog.V(2).Infof("ImageConfig 'cluster' does not exist or has been deleted")
 		return nil
 	}
 	if err != nil {
 		return err
 	}
-
 	// Deep-copy otherwise we are mutating our cache.
 	imgcfg = imgcfg.DeepCopy()
 
+	// Fetch the ClusterVersionConfig needed to get the registry being used by the payload
+	// so that we can avoid adding that registry to blocked registries in /etc/containers/registries.conf
+	clusterVersionCfg, err := ctrl.clusterVersionLister.Get("version")
+	if errors.IsNotFound(err) {
+		glog.Infof("ClusterVersionConfig 'version' does not exist or has been deleted")
+		return nil
+	}
+	if err != nil {
+		return err
+	}
+	// Deep-copy otherwise we are mutating our cache
+	clusterVersionCfg = clusterVersionCfg.DeepCopy()
```

Comment:
You don't need to deepcopy, you're not modifying the object at all

Reply:
Fixed, thanks!